Gems/ONNX/Code/Source/Clients/Model.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ONNX/Model.h>
+
+namespace ONNX
+{
+    void Model::Load(const InitSettings& initSettings)
+    {
+        // Get the FileIOBase to resolve the path to the ONNX gem
+        AZ::IO::FixedMaxPath onnxModelPath;
+
+        // If no filepath provided for onnx model, set default to a model.onnx file in the Assets folder.
+        if (initSettings.m_modelFile.empty())
+        {
+            AZ::IO::FileIOBase* fileIo = AZ::IO::FileIOBase::GetInstance();
+            fileIo->ResolvePath(onnxModelPath, "@gemroot:ONNX@/Assets/model.onnx");
+        }
+        else
+        {
+            onnxModelPath = initSettings.m_modelFile;
+        }
+
+        // If no model name is provided, will default to the name of the onnx model file.
+        if (initSettings.m_modelName.empty())
+        {
+            AZ::StringFunc::Path::GetFileName(onnxModelPath.c_str(), m_modelName);
+        }
```

Comment:
This should use `onnyModelPath.Filename()` to retrieve the the filename part of the path.
The `StringFunc::Path` utilities are deprecated for the AZ::IO::Path facilities.
```suggestion
        m_modelName = onnyModelPath.Filename().String();
```

Reply:
Changed